### PR TITLE
Add initial federation SCIM plugin

### PR DESCRIPTION
**Pull request by justin-stephenson**
New federation plugin to replace `sssd` plugin, ported from https://github.com/justin-stephenson/scim-keycloak-user-storage-spi/

Implementation details: https://github.com/justin-stephenson/scim-keycloak-user-storage-spi/?tab=readme-ov-file#plugin-communication

### CODE REVIEW

**Comment by jonkoops** [comment]
Thanks Justin, this is really valuable. We're currently working hard on the v26 release, so it might take a little before we get time to take a look at this.

File: federation/scim/src/main/java/org/keycloak/scim_user_spi/SCIMUserStorageProviderFactory.java
Comment (stianst) [inline federation/scim/src/main/java/org/keycloak/scim_user_spi/SCIMUserStorageProviderFactory.java:1]: We should put an experimental feature guard so it's not enabled/supported when first merged.

**Review by stianst — COMMENTED** [review]
I will talk to @pedroigor next week to figure out when the team can review this in more detail. Hopefully though we can get this merged as a experimental feature shortly after Keycloak 26 is out.

Some things we need to figure out:

* How to handle long response time when creating the provider initially when `Add integration domain` is enabled. We can't require folks to bump the timeout for http requests, and it would also be rather bad user experience if the admin console basically freezes for 30 seconds. Most likely someone will press the refresh button a few times by then ;)
* We need to figure out how will tests this on our end. We can discuss this further once the team can get involved more

We probably also want to plan updates in a few rounds:

1. Get the PR merged as an experimental feature - this can happen relatively quickly, and doesn't need tests, docs, etc.
2. Plan out what is needed to do before we can make it supported as an MVP feature
3. Plan what we want to do as follow-ups. Things like mappers is probably fine to do as an MVP. Other things like re-using this code to make a generic SCIM provider as well.

**Comment by justin-stephenson** [comment]
@pedroigor I renamed the plugin and references from `scim` to `ipatuura` as we discussed, and rebased the PR. Feel free to reach out if you have any more questions or for help with setting up the plugin + ipatuura.

**Comment by pedroigor** [comment]
@justin-stephenson Thanks. We are working on it now, @sguilhen should get in touch soon with more updates.

**Comment by sguilhen** [comment]
Ah yes, sorry for that :smile:

**Pull request closed without merging**